py_cui/control_widgets/slider.py

```diff
@@ -74,51 +89,147 @@ class SliderWidget(py_cui.widgets.Widget, SliderImplementation):
     def __init__(self, id, title, grid, row, column, row_span, column_span,
                  padx, pady, logger, min_val, max_val, step, init_val):
 
-        SliderImplementation.__init__(self, min_val, max_val, init_val, step)
+        SliderImplementation.__init__(self, min_val, max_val, init_val, step, logger)
 
         py_cui.widgets.Widget.__init__(self, id, title, grid, row, column,
                                        row_span, column_span, padx,
                                        pady, logger, selectable=True)
 
+        self._title_enabled = False
+        self._border_enabled = False
+        self._display_value = True
+        self._alignment = "mid"
+        self.set_help_text("Focus mode on Slider. Use left/right to adjust value. Esc to exit.")
+
+
+    def toggle_title(self):
+        """Toggles visibility of the widget's name.
+        """
+
+        self._title_enabled = not self._title_enabled
+
+
+    def toggle_border(self):
+        """Toggles visibility of the widget's border. Enabling this will disable the alignment.
+        """
+
+        self._border_enabled = not self._border_enabled
+
+
+    def toggle_value(self):
+        """Toggles visibility of the widget's current value in integer.
+        """
+
+        self._display_value = not self._display_value
+
+
+    def align_to_top(self):
+        """Aligns widget height to top.
+        """
+        self._alignment = "top"
+
+
+    def align_to_middle(self):
+        """Aligns widget height to middle, default option.
+        """
+        self._alignment = "mid"
+
+
+    def align_to_bottom(self):
+        """Aligns widget height to bottom.
+        """
+        self._alignment = "btm"
+
+
+    def _custom_draw_border(self, start_y: int):
+        """Custom method made from renderer.draw_border to support alignment for bordered variants.
+
+        Parameters
+        ----------
+        start_y : int
+            border's Y-axis starting coordination
+        """
+
+        start_y = start_y
```

Comment:
I don't think this line is necessary?

Reply:
Oh, I refactored names and forgot to remove those. I'll do it right away. Nice catch!

Reply:
If you are interested in further contributions, the current list of issues I want to fit in before `v0.1.3` is here: https://github.com/jwlodek/py_cui/milestone/8. I might decide to not include some since a lot has been added since `v0.1.2` already.

Reply:
I do remember seeing various changes on `file-explorer-widget` branch. Since it's merged, it is safe to assume that the master is the most up-to-date branch, including local branches?

Since my tiny project does multiple access to protected attributes of ```ScrollWidget``` for some features, I'd like to make a issue for those and work on it. 

Things I have interest with are as follows, and wishing those not to cause merge conflicts, tell me if you're already working on it!

- ```list.clear()``` instead of reassignment - [reference](https://stackoverflow.com/questions/50895560)
- Public access to items - https://github.com/jwlodek/py_cui/issues/61#issuecomment-774675964
- Callback on selection changes

Reply:
I already wrapped up #61, the other two you can work on for sure. Probably they will wait to be merged till after v0.1.3, since it's been a while since the last release and I want to avoid feature creep.